lib/src/solver/incompatibility.dart

```diff
@@ -419,9 +419,9 @@ class Incompatibility {
       }
     } else if (latter.cause == IncompatibilityCause.noVersions) {
       buffer.write("which doesn't match any versions");
-    } else if (cause is PackageNotFoundCause) {
+    } else if (latter.cause is PackageNotFoundCause) {
       buffer.write("which doesn't exist "
-          '(${(cause as PackageNotFoundCause).exception.message})');
+          '(${(latter.cause as PackageNotFoundCause).exception.message})');
```

Comment:
I agree that it is very hard to follow the logic here.

But just pattern matching with the lines above, your version seems to be the right thing.

The code adding that case did not add any test: https://github.com/dart-lang/pub/commit/dcada10a0ac984058f79a85677a52d7b2343ed6d
So it is very reasonable that it could be wrong.